kale/loaddata/dataset_access.py

```diff
@@ -28,21 +28,49 @@ def get_train(self):
         """
         raise NotImplementedError()
 
-    def get_train_val(self, val_ratio):
+    def get_class_subsampled_train(self, class_ids=None):
+        """
+        Args:
+            class_ids (list, optional): List of class ids that are to be subsampled.
+        Returns: a torch.utils.data.Dataset
+            Dataset: a torch.utils.data.Dataset with only classes in class_ids
+        """
+        train_dataset = self.get_train()
+        if class_ids is None:
+            return train_dataset
+        else:
+            sub_indices = [i for i in range(0, len(train_dataset)) if train_dataset[i][1] in class_ids]
+            return torch.utils.data.Subset(train_dataset, sub_indices)
+
+    def get_train_val(self, val_ratio, sub_class_ids=None):
         """
         Randomly split a dataset into non-overlapping training and validation datasets.
 
         Args:
             val_ratio (float): the ratio for validation set
-
+            sub_class_ids (list, optional): List of class ids that are to be subsampled.
         Returns:
             Dataset: a torch.utils.data.Dataset
         """
-        train_dataset = self.get_train()
+        train_dataset = self.get_class_subsampled_train(sub_class_ids)
         ntotal = len(train_dataset)
         ntrain = int((1 - val_ratio) * ntotal)
 
         return torch.utils.data.random_split(train_dataset, [ntrain, ntotal - ntrain])
 
     def get_test(self):
         raise NotImplementedError()
+
+    def get_class_subsampled_test(self, class_ids):
+        """
+        Args:
+            class_ids (list, optional): List of class ids that are to be subsampled.
+        Returns: a torch.utils.data.Dataset
+            Dataset: a torch.utils.data.Dataset with only classes in class_ids
+        """
+        test_dataset = self.get_test()
+        if class_ids is None:
+            return test_dataset
+        else:
+            sub_indices = [i for i in range(0, len(test_dataset)) if test_dataset[i][1] in class_ids]
+            return torch.utils.data.Subset(test_dataset, sub_indices)
```

Comment:
Line 72-76 is somehow a repetition of line 39-43. Is it possible and worthy to reduce such repetition (e.g. define that 4 lines as a function taking ids and dataset in)?

Reply:
Yes I feel that is definitely worthy. I am thinking a private _get_subset function? Also, I wanted to ask whether this would be alright since this function will come in the 'read the docs' as well.

Reply:
Code will be reviewed and can be updated so you may make the changes good in your opinion and we will review. When not sure, check how pytorch implements something similar to learn.